apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -299,12 +297,20 @@ impl PlannerMode {
                 let (plan, mut root_node) = tokio::task::spawn_blocking(move || {
                     let start = Instant::now();
 
+                    let query_plan_options = QueryPlanOptions {
+                        override_conditions: HashSet::from_iter(plan_options.override_conditions),
+                    };
+
                     let result = operation
                         .as_deref()
                         .map(|n| Name::new(n).map_err(FederationError::from))
                         .transpose()
                         .and_then(|operation| {
-                            rust_planner.build_query_plan(&doc.executable, operation)
+                            rust_planner.build_query_plan(
+                                &doc.executable,
+                                operation,
+                                Some(query_plan_options),
+                            )
```

Comment:
Plan options are a `Vec` in the router bridge rather than an `Option`. Given that these are ... options ... an `Option` does seem to make more sense, but we are certain we are okay with the different API here?

Reply:
Yeah will drop option (see my other comment) and change from hashset to vec (on options) for simplicity.